awscli/customizations/s3events.py

```diff
@@ -57,6 +65,46 @@ def replace_event_stream_docs(help_command, **kwargs):
               "object content is written to the specified outfile.\n")
 
 
+def document_expires_string(help_command, **kwargs):
+    doc = help_command.doc
+    caught_index_error = False
+    popped = []
+    current = ''
+    while current != 'Expires -> (timestamp)':
+        try:
+            current = doc.pop_write()
+            popped.append(current)
+        except IndexError:
+            # Do nothing if Expires is not in the modeled response
+            caught_index_error = True
+            break
+
+    # Only inject the deprecation notice if the Expires timestamp arg was found in the document
+    if not caught_index_error:
+        # Put back the expires field and description
+        doc.push_write(popped.pop())
```

Comment:
This depends on the precondition that each field of the output body is made up of exactly 4 items on the documentation stack. This appears true as of writing. 

I believe this feels convoluted, and I prefer botocore's use of the DocumentStructure class, which makes it easier to inject notices like this. It would take non-trivial work to migrate over to use DocumentStructure, and it might not be worth the effort.